django/contrib/gis/gdal/raster/source.py

```diff
@@ -443,6 +462,9 @@ def transform(self, srs, driver=None, name=None, resampling='NearestNeighbour',
                 'Transform only accepts SpatialReference, string, and integer '
                 'objects.'
             )
+
+        if target_srs.srid == self.srid:
```

Comment:
The transform function is not only to re-project a raster. It can also be used to change its driver (by providing the same srs but a different driver name than the original). Admitedly, for that purpose the `warp` function might be more appropriate, but people might be using transform as well as "filetype change function" convenience. So before going into "clone mode", we might have to check for the equal drivers as well? I.e. `if diver is not None and driver == self.driver.name` or something along those lines.

Reply:
Fixed.